python/pyspark/ml/param/__init__.pyi

```diff
@@ -94,3 +94,6 @@ class Params(pyspark.ml.util.Identifiable, metaclass=abc.ABCMeta):
     def copy(self, extra: Optional[pyspark.ml._typing.ParamMap] = ...) -> Params: ...
     def set(self, param: Param, value: Any) -> None: ...
     def clear(self, param: Param) -> None: ...
+    def _setDefault(self: P, **kwargs: Any) -> P: ...
+    @staticmethod
+    def _dummy() -> Params: ...
```

Comment:
just for doubly sure, this is to avoid mypy complaints right?

Reply:
That's correct. We use `Params._dummy` as a `parent` for newly initialized `Param`.  It makes more sense to fix it here (it will be visible once we inline `__init__.py`, than adding more cleanup work later.